src/border_agent/border_agent.cpp

```diff
@@ -211,6 +218,30 @@ void BorderAgent::PublishMeshCopService(void)
     // "xa" stands for Extended MAC Address (64-bit) of the Thread Interface of the Border Agent.
     txtList.emplace_back("xa", extAddr->m8, sizeof(extAddr->m8));
 
+#if OTBR_ENABLE_BORDER_ROUTING
+    while (otNetDataGetNextOnMeshPrefix(instance, &iterator, &config) == OT_ERROR_NONE)
+    {
+        if (otBorderRouterIsValidOmrPrefix(instance, &config))
+        {
+            if (!hasOmr || memcmp(config.mPrefix.mPrefix.mFields.m8, smallestOmrPrefix.mPrefix.mFields.m8,
+                                  OTBR_IP6_PREFIX_SIZE) < 0)
+            {
+                hasOmr            = true;
+                smallestOmrPrefix = config.mPrefix;
+            }
+        }
+    }
+
+    if (hasOmr)
+    {
+        std::vector<uint8_t> value(sizeof(uint8_t) + OT_IP6_PREFIX_SIZE);
+        value[0] = smallestOmrPrefix.mLength;
+        std::copy(smallestOmrPrefix.mPrefix.mFields.m8, smallestOmrPrefix.mPrefix.mFields.m8 + OT_IP6_PREFIX_SIZE,
+                  value.begin() + 1);
+        txtList.emplace_back("omr", value.data(), value.size());
+    }
+#endif
```

Comment:
Write a small function for this?

Reply:
We may add a method to ControllerOpenThread?